client/src/index.tsx

```diff
@@ -48,22 +46,21 @@ const link = split(
 export const client = new ApolloClient({
     link,
     cache: new InMemoryCache({
-        dataIdFromObject: (object: any) => {
-            switch (object.__typename) {
-                case 'Location':
-                    return "Location:" + object.location_id;
-                case 'Category':
-                    return "Category:" + object.category_id; // use `category_id` as the primary key
-                case 'User':
-                    return "User:" + object.uuid; // use `uuid` as the primary key
-                case 'Request':
-                    return "Request:" + object.request_id; // use `request_id` as the primary key
-                case 'Item':
-                    return "Item:" + object.id;
-                case 'Setting':
-                    return "Setting:" + object.name;
-                default:
-                    return defaultDataIdFromObject(object); // fall back to default handling
+        typePolicies: { // specify custom primary keys for certain types for caching
```

Comment:
THIS IS SO MUCH BETTER!! thank you Apollo v3 😊 

I fought Apollo so much when I originally implemented this

Reply:
Yes, it's definitely much cleaner haha